components/add_groups_to_team_modal/add_groups_to_team_modal.jsx

```diff
@@ -0,0 +1,278 @@
+                    values={{
+                        num: MAX_SELECTABLE_VALUES - this.state.values.length,
+                    }}
+                />
+            </div>
+        );
+
+        const buttonSubmitText = localizeMessage('multiselect.add', 'Add');
+        const buttonSubmitLoadingText = localizeMessage('multiselect.adding', 'Adding...');
+
+        let groups = [];
+        if (this.props.groups) {
+            groups = this.props.groups.filter((user) => user.delete_at === 0);
+        }
+
+        let addError = null;
+        if (this.state.addError) {
+            addError = (<div className='has-error col-sm-12'><label className='control-label font-weight--normal'>{this.state.addError}</label></div>);
+        }
+
+        return (
+            <Modal
+                id='addUsersToTeamModal'
+                dialogClassName={'more-modal more-direct-channels'}
+                show={this.state.show}
+                onHide={this.handleHide}
+                onExited={this.handleExit}
+            >
+                <Modal.Header closeButton={true}>
+                    <Modal.Title>
+                        <FormattedMessage
+                            id='add_groups_to_team.title'
+                            defaultMessage='Add New Groups To {teamName} Team'
+                            values={{
+                                teamName: (
+                                    <strong>{this.props.currentTeamName}</strong>
+                                ),
+                            }}
+                        />
+                    </Modal.Title>
+                </Modal.Header>
+                <Modal.Body>
+                    {addError}
+                    <MultiSelect
+                        key='addGroupsToTeamKey'
+                        options={groups}
+                        optionRenderer={this.renderOption}
+                        values={this.state.values}
+                        valueRenderer={this.renderValue}
+                        perPage={GROUPS_PER_PAGE}
+                        handlePageChange={this.handlePageChange}
+                        handleInput={this.search}
+                        handleDelete={this.handleDelete}
+                        handleAdd={this.addValue}
+                        handleSubmit={this.handleSubmit}
+                        maxValues={MAX_SELECTABLE_VALUES}
+                        numRemainingText={numRemainingText}
+                        buttonSubmitText={buttonSubmitText}
+                        buttonSubmitLoadingText={buttonSubmitLoadingText}
+                        saving={this.state.saving}
+                        loading={this.state.loadingGroups}
+                        placeholderText={localizeMessage('multiselect.addGroupsPlaceholder', 'Search and add groups')}
+                    />
+                </Modal.Body>
+            </Modal>
+        );
+    }
+}
```

Comment:
This component is a fairly straight copy of components/add_users_to_team/add_users_to_team.jsx.